lib/tapioca/gem/listeners/not_owned_constants.rb

```diff
@@ -0,0 +1,32 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Tapioca
+  module Gem
+    module Listeners
+      class NotOwnedConstants < Base
+        extend T::Sig
+
+        include Runtime::Reflection
+
+        private
+
+        sig { override.params(event: ScopeNodeAdded).void }
+        def on_scope(event)
+          mixin = event.constant
+
+          # There are cases where we want to compile RBI for constants not owned by the current
+          # gem - e.g. constants defined in one gem that then have a mixin applied to them in
+          # another gem. This is especially common for Ruby standard library classes.
+          #
+          # We can identify these cases by keeping track of which classes have mixins, and then
+          # adding those classes to the pipeline.
```

Comment:
This comment is correct but I think we can improve it slightly. What do you think of this:
```suggestion
          # There are cases where we want to process constants not declared by the current
          # gem, i.e. "foreign constant". These are constants defined in another gem 
          # that this gem is applying a mix-in to. This pattern is especially common 
          # for gems that add behaviour to Ruby standard library classes by mixing in
          # modules to them.
          #
          # The way we identify these "foreign constants" is by asking the mixin
          # tracker which constants have mixed in the current module that we are
          # handling. We add all those constants that we discover to the pipeline 
          # to be processed.
```